Frameworks/CoreGraphics/CGGraphicBufferImage.mm

```diff
@@ -30,9 +40,140 @@
 
 #pragma clang diagnostic pop
 
+using namespace Microsoft::WRL;
+
 static const wchar_t* TAG = L"CGGraphicBufferImage";
 extern int imgDataCount;
 
+/**
+ * Custom Core Graphic IWICBitmap interface that uses the backing image buffer without copying.
+ */
+
+class CGGraphicBufferWICBitmap;
+
+class CGGraphicBufferWICBitmapLock : public RuntimeClass<Microsoft::WRL::RuntimeClassFlags<Microsoft::WRL::RuntimeClassType::ClassicCom>,
+                                                         IAgileObject,
+                                                         Microsoft::WRL::FtmBase,
+                                                         IWICBitmapLock> {
+public:
+    CGGraphicBufferWICBitmapLock();
+
+    HRESULT RuntimeClassInitialize(_In_ CGGraphicBufferWICBitmap* imageWICBitmap);
+
+    HRESULT STDMETHODCALLTYPE GetSize(_Out_ UINT* width, _Out_ UINT* height);
+    HRESULT STDMETHODCALLTYPE GetStride(_Out_ UINT* stride);
+    HRESULT STDMETHODCALLTYPE GetDataPointer(_Out_ UINT* bufferSize, _Outptr_ WICInProcPointer* data);
+    HRESULT STDMETHODCALLTYPE GetPixelFormat(_Out_ WICPixelFormatGUID* pixelFormat);
+
+private:
+    CGGraphicBufferWICBitmap* m_imageWICBitmap;
+};
+
+class CGGraphicBufferWICBitmap : public RuntimeClass<Microsoft::WRL::RuntimeClassFlags<Microsoft::WRL::RuntimeClassType::ClassicCom>,
+                                                     IAgileObject,
+                                                     Microsoft::WRL::FtmBase,
+                                                     IWICBitmap> {
+public:
+    CGGraphicBufferWICBitmap();
+
+    HRESULT RuntimeClassInitialize(
+        _In_ UINT width, _In_ UINT height, _In_ UINT stride, _In_ WICPixelFormatGUID pixelFormat, _In_ BYTE* dataBuffer);
+
+    // IWICBitmap interface
+    HRESULT STDMETHODCALLTYPE Lock(_In_ const WICRect* lockRect, _In_ DWORD flags, _COM_Outptr_ IWICBitmapLock** outLock) {
+        ComPtr<IWICBitmapLock> lock;
+        MakeAndInitialize<CGGraphicBufferWICBitmapLock>(&lock, this);
+        *outLock = lock.Detach();
+        return S_OK;
+    }
+
+    HRESULT STDMETHODCALLTYPE SetPalette(_In_ IWICPalette* palette) {
+        return E_NOTIMPL;
+    }
+
+    HRESULT STDMETHODCALLTYPE SetResolution(_In_ double dpiX, _In_ double dpiY) {
```

Comment:
>SetResolution [](start = 30, length = 13)

Does this ever get called?  It would be useful to add UNIMPLEMENTED() to these not implemented methods.